File: BlackBerry_Dynamics_SDK_for_Cordova_v11.0.0.7/plugins/cordova-plugin-bbd-launcher/scripts/hooks/rmLauncherFramework.rb
#!/usr/bin/ruby

#
# (c) 2021 BlackBerry Limited. All rights reserved.
#

begin
  require_relative './launcherManager.rb'

  manager = LauncherPluginManager.new
  manager.rm_framework
rescue => e
  ErrorManager.print_exception e
  exit 1
end

File: BlackBerry_Dynamics_SDK_for_Cordova_v11.0.0.7/plugins/cordova-plugin-bbd-launcher/scripts/hooks/rmLauncherFramework.js
#!/usr/bin/env node

/*
 * (c) 2021 BlackBerry Limited. All rights reserved.
 */

'use strict';

const path = require('path');
require('shelljs/global');

module.exports = function (context) {
  if (exec(`ruby "${path.join(__dirname, 'rmLauncherFramework.rb')}"`).code !== 0) {
    throw new Error('\nERROR: rmLauncherFramework exited with error!');
  }
};
